ethereum/nft/trades/insert_rarible.sql

```diff
@@ -0,0 +1,362 @@
+    FROM rarible."ExchangeV1_evt_Buy"
+    where "buyTokenId" = 0 --buy
+    UNION ALL
+-- from Nov 2020 ongoing
+    SELECT
+        'Rarible',
+        '1',
+        '\xcd4ec7b66fbc029c116ba9ffb3e59351c20b5b06',
+        'Trade',
+        evt_tx_hash,
+        evt_block_time,
+        evt_block_number,
+        evt_index,
+        "buyToken",
+        CAST( "buyTokenId" AS TEXT),
+        buyer AS seller,
+        owner AS buyer,
+        amount,
+        "sellToken",
+        CASE
+            WHEN "sellToken" = '\x0000000000000000000000000000000000000000' THEN '\xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2'
+            ELSE "sellToken"
+        END,
+        'ExchangeV1_evt_Buy' as category
+    FROM rarible."ExchangeV1_evt_Buy"
+    where "sellTokenId" = 0 
+    UNION ALL
+-- from 2021-06-15 onwards
+-- 'Purchases' in ETH
+    SELECT
+        'Rarible' as platform,
+        '2' as platform_version,
+        contract_address AS exchange_contract_address,
+        'Trade' as evt_type,
+        tx_hash AS evt_tx_hash,
+        block_time AS evt_block_time,
+        block_number AS evt_block_number,
+        "index" AS evt_index,
+        substring(data FROM 365 FOR 20) AS nft_contract_address,
+        CAST(bytea2numericpy(substring(data FROM 385 FOR 32)) AS TEXT) AS nft_token_id,
+        substring(data FROM 77 FOR 20) AS seller,
+        substring(data FROM 109 FOR 20) AS buyer,
+        bytea2numericpy(substring(data FROM 129 FOR 32)) original_amount_raw,
+        '\x0000000000000000000000000000000000000000'::bytea as original_currency_contract,
+        '\xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2'::bytea as currency_contract,
+        'Buy' AS category
+    FROM ethereum."logs" 
+    WHERE "contract_address" = '\x9757f2d2b135150bbeb65308d4a91804107cd8d6' 
+    AND topic1 = '\x268820db288a211986b26a8fda86b1e0046281b21206936bb0e61c67b5c79ef4'
+    AND length(data) = 512
+UNION ALL
+-- from 2021-06-15 onwards
+-- 'Bid Accepted' non-ETH
+    SELECT
+        'Rarible' as platform,
+        '2' as platform_version,
+        contract_address AS exchange_contract_address,
+        'Trade' as evt_type,
+         tx_hash AS evt_tx_hash,
+        block_time AS evt_block_time,
+        block_number AS evt_block_number,
+        "index" AS evt_index,
+        substring(data FROM 493 FOR 20) AS nft_contract_address,
+        CAST(bytea2numericpy(substring(data FROM 513 FOR 32)) AS TEXT) AS nft_token_id,
+        substring(data FROM 109 FOR 20) AS seller,
+        substring(data FROM 77 FOR 20) AS buyer,
+        bytea2numericpy(substring(data FROM 161 FOR 32)) AS original_amount_raw,
+        substring(data FROM 365 FOR 20) AS original_currency_contract,
+        substring(data FROM 365 FOR 20) AS currency_contract,
+        'Offer Accepted' AS category
+    FROM ethereum."logs"
+    WHERE "contract_address" = '\x9757f2d2b135150bbeb65308d4a91804107cd8d6'
+    AND topic1 = '\x268820db288a211986b26a8fda86b1e0046281b21206936bb0e61c67b5c79ef4'
```

Comment:
The above 2 selects look similar. The main problem with the trade_ids is that if any of these selects in this union chain return the same trace of event log it will get recorded twice which is bad.